src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.CodeModification/CodeModification/ProjectModifier.cs

```diff
@@ -122,6 +128,29 @@ public async Task<bool> RunAsync()
         return await ProjectModifierHelper.ModifyDocumentTextAsync(document, replacements);
     }
 
+    /// <summary>
+    /// Updates .razor and .html files via string replacement
+    /// </summary>
+    /// <param name="file"></param>
+    /// <param name="toolOptions"></param>
+    /// <returns></returns>
+    internal static async Task<Document?> ApplyTextReplacements(CodeFile file, Document? document, IList<string> toolOptions)
+    {
+        if (document is null)
+        {
+            return null;
+        }
+
+        var replacements = file.Replacements?.Where(cc => ProjectModifierHelper.FilterOptions(cc.Options, toolOptions));
+        if (replacements is null || !replacements.Any())
+        {
+            return document;
+        }
+
+        //should be a safe cast as we are starting with a 'Document' type
```

Comment:
yep this seems way easier, I was just rushing this change so didn't think it through

Reply:
also switched to `Task<T?> ApplyTextReplacements<T>` instead of having multiple definitions of this method for both Document types. Looks and works good.